public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -45,6 +47,89 @@ export class NodePosition {
     return query.substring(this.from, this.to);
   }
 }
+
+/**
+ * Checks for the presence of a given label=value filter in any Matcher expression in the query.
+ */
+export function queryHasFilter(query: string, key: string, operator: string, value: string): boolean {
+  const matchers = getMatchersWithFilter(query, key, operator, value);
+  return matchers.length > 0;
+}
+
+/**
+ * Removes a label=value Matcher expression from the query.
+ */
+export function removeLabelFromQuery(query: string, key: string, operator: string, value: string): string {
+  const matchers = getMatchersWithFilter(query, key, operator, value);
+  for (const matcher of matchers) {
+    query =
+      matcher.parent?.type.id === LabelFilter ? removeLabelFilter(query, matcher) : removeSelector(query, matcher);
+  }
+  return query;
+}
+
+function removeLabelFilter(query: string, matcher: SyntaxNode): string {
+  const pipelineStage = matcher.parent?.parent;
+  if (!pipelineStage) {
+    return query;
+  }
+  return (query.substring(0, pipelineStage.from) + query.substring(pipelineStage.to)).trim();
+}
+
+function removeSelector(query: string, matcher: SyntaxNode): string {
+  let selector: SyntaxNode | null = matcher;
+  do {
+    selector = selector.parent;
+  } while (selector && selector.type.id !== Selector);
+  const label = matcher.getChild(Identifier);
+  if (!selector || !label) {
+    return query;
+  }
+  const labelName = query.substring(label.from, label.to);
+  const modeller = new LokiQueryModeller();
```

Comment:
It requires an instance.

Reply:
Yea, sorry, bad phrasing. What I meant is: do we need to recreate the instance in every `removeSelector` selector call, or can the instance be outside of the method?

Reply:
My bad, because it definitely looks like something where it could all be static methods or just functions, there's not real need for new instances.

TIL the module itself exports a single instance, so I figured we can use that instead: https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/querybuilder/LokiQueryModeller.ts#L198